SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -92,23 +102,35 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		return;
 	}
 
-	//checkf(!SpatialOSInstance->IsConnected(), TEXT("SpatialOS should not be connected already. This is probably because we attempted to travel to a different level, which current isn't supported."));
+	if (Connection)
+	{
+		// If the connection already exists in this NetDriver we don't need to continue this function.
+		return;
+	}
+
+	// Bind the ProcessServerTravel delegate to the spatial variant. This ensures that if ServerTravel is called and Spatial networking is enabled, we can travel properly.
+	LoadedWorld->SpatialProcessServerTravelDelegate.BindStatic(SpatialProcessServerTravel);
+
+	// Handle Spatial connection configurations.
+	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
 
 	// Set the timer manager.
 	TimerManager = &LoadedWorld->GetTimerManager();
 
 	// Set up manager objects.
 	EntityRegistry = NewObject<UEntityRegistry>(this);
 
-	// Handle Spatial connection configurations.
-	UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Loaded Map %s. Connecting to SpatialOS."), *LoadedWorld->GetName());
-
-	if (Connection != nullptr)
+	if (!bPersistSpatialConnection)
 	{
-		return;
+		// Destroy the old connection
+		Cast<USpatialGameInstance>(GetWorld()->GetGameInstance())->SpatialConnection->DestroyConnection();
```

Comment:
Nit: Could you make `Cast<USpatialGameInstance>(GetWorld()->GetGameInstance())` into a local variable right before
`if (!bPersistSpatialConnection)`?